pkg/quota/admission/clusterresourceoverride/admission.go

```diff
@@ -0,0 +1,161 @@
+package clusterresourceoverride
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"reflect"
+
+	oadmission "github.com/openshift/origin/pkg/cmd/server/admission"
+	"github.com/openshift/origin/pkg/project/cache"
+	"github.com/openshift/origin/pkg/quota/admission/clusterresourceoverride/api"
+	"github.com/openshift/origin/pkg/quota/admission/clusterresourceoverride/api/validate"
+	"k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/plugin/pkg/admission/limitranger"
+
+	"github.com/ghodss/yaml"
+	"github.com/golang/glog"
+	"speter.net/go/exp/math/dec/inf"
+)
+
+const (
+	clusterResourceOverrideAnnotation = "quota.openshift.io/cluster-resource-override-enabled"
+	cpuBaseScaleFactor                = 1000.0 / (1024.0 * 1024.0 * 1024.0) // 1000 milliCores per 1GiB
+)
+
+func init() {
+	admission.RegisterPlugin("ClusterResourceOverride", func(client kclient.Interface, config io.Reader) (admission.Interface, error) {
+		return newClusterResourceOverride(client, config)
+	})
+}
+
+type clusterResourceOverridePlugin struct {
+	*admission.Handler
+	Config       *api.ClusterResourceOverrideConfig
+	ProjectCache *cache.ProjectCache
+	LimitRanger  admission.Interface
+}
+
+var _ = oadmission.WantsProjectCache(&clusterResourceOverridePlugin{})
+var _ = oadmission.Validator(&clusterResourceOverridePlugin{})
+
+// newClusterResourceOverride returns an admission controller for containers that
+// configurably overrides container resource request/limits
+func newClusterResourceOverride(client kclient.Interface, config io.Reader) (admission.Interface, error) {
+	glog.V(5).Infof("ClusterResourceOverride admission controller is loaded")
+	parsed, err := ReadConfig(config)
+	glog.V(5).Infof("ClusterResourceOverride admission controller got config: %v\nerror: (%T) %[2]v", parsed, err)
+	return &clusterResourceOverridePlugin{
+		Handler:     admission.NewHandler(admission.Create),
+		Config:      parsed,
+		LimitRanger: limitranger.NewLimitRanger(client, wrapLimit),
```

Comment:
This is adding another list/watch and cache instead of re-using the existing one.  In addition, it looks like this admission plugin logically requires the other one to activated.  Is there a reasonable way to wire these together?  @derekwaynecarr  is the too expensive to allow?  I'm inclined to allow a followup issue unless you think this is really problematic.

I see two followups:
 1. Some additional `Wants` interface that can say that you must have some sort of admission plugin.  The list should be split for before and after in the chain.
 1. Share a limit ranger or at least the cache to avoid coherence problems.